### PR TITLE
[ci skip] Refactored Node and Node Registration Roles

### PR DESCRIPTION
**Pull request by sauveergoel**
Signed-off-by: sauveergoel <sauveer.goel@accenture.com>

**Changelog**
- Refactored Node and Node Registration Roles

**Reviewed by**
@jagpreetsinghsasan @abevers @suvajit-sarkar @sownak 

**Linked issue**
#976

### CODE REVIEW

File: platforms/corda-ent/configuration/roles/helm_component/templates/node.tpl [hunk 1]
```diff
@@ -28,48 +28,48 @@ spec:
...
       doormanURL: {{ doorman_url }}
       networkMapURL: {{ networkmap_url }}
-      idmanDomain: {{ idman_domain }}
-      networkMapDomain: {{ networkmap_domain }}
+      idmanDomain: "{{ doorman_url.split(':')[1] | regex_replace('/', '') }}"
+      networkMapDomain: "{{ networkmap_url.split(':')[1] | regex_replace('/', '') }}"
     firewall:
-      enabled: {{ firewall_enabled }}
+      enabled: {{ peer.firewall.enabled }}
     cenmServices:
```
Comment (sownak) [inline platforms/corda-ent/configuration/roles/helm_component/templates/node.tpl:35]: cenmservices to be merged with networkservices

**Review by sownak — CHANGES_REQUESTED** [review]
The chart values.yaml need to be refacotored first and then these changes should have been made. I do not see any chart changes in this PR

File: platforms/corda-ent/configuration/roles/helm_component/templates/node.tpl (hunk 1 above)
Comment (sauveergoel) [reply platforms/corda-ent/configuration/roles/helm_component/templates/node.tpl]: done

File: platforms/corda-ent/charts/node-initial-registration/values.yaml [hunk 2]
```diff
@@ -65,41 +65,48 @@ vault:
...
+  devMode: false
+  # Provide volume related specifications
+  volume:
+    # Ex baseDir: /opt/corda
+    baseDir:
+  # where is node jar is stored
+  jarPath: bin
+  # where service configuration files are stored
+  configPath: etc
+  cordaJarMx:
```
Comment (sownak) [inline platforms/corda-ent/charts/node-initial-registration/values.yaml:98]: Add comments from platforms/corda-ent/charts/notary-initial-registration/values.yaml

Comment (sownak) [reply platforms/corda-ent/charts/node-initial-registration/values.yaml:98]: As well as the correct memory parameters

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/corda-ent/charts/node-initial-registration/values.yaml [hunk 3]
```diff
@@ -109,68 +116,59 @@ service:
...
-volume:
-  # Ex baseDir: /opt/corda
-   baseDir:
-# where is node jar is stored
-jarPath: bin
-# where service configuration files are stored
-configPath: etc
-
-cordaJarMx: 
+#############################################################
```
[the remark this answers is not in the document]
Comment (sauveergoel) [reply platforms/corda-ent/charts/node-initial-registration/values.yaml:202]: done changes

File: platforms/corda-ent/charts/node-initial-registration/values.yaml (hunk 2 above)
Comment (sauveergoel) [reply platforms/corda-ent/charts/node-initial-registration/values.yaml]: done changes

File: platforms/corda-ent/charts/node-initial-registration/templates/job.yaml [hunk 4]
```diff
@@ -66,13 +66,13 @@ spec:
...
                 COUNTER=1
-                while [ "$COUNTER" -lt {{ $.Values.healthcheck.readinessthreshold }} ]
+                while [ "$COUNTER" -lt {{ $.Values.vault.retries }} ]
                 do
                     # get keystores from vault to see if certificates are created and put in vault
                     LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${CERTS_SECRET_PREFIX}/root/certs | jq -r 'if .errors then . else . end')
                     if echo ${LOOKUP_SECRET_RESPONSE} | grep "errors"
                     then
-                      sleep {{ $.Values.healthcheck.readinesscheckinterval }}
+                      sleep {{ $.Values.dataSourceProperties.retryInterval }}
```
Comment (sownak) [inline platforms/corda-ent/charts/node-initial-registration/templates/job.yaml:75]: shouldnt this be vault.retryinterval?

Comment (sauveergoel) [reply platforms/corda-ent/charts/node-initial-registration/templates/job.yaml]: done changes

**Pull request merged**